session/pessimistic_test.go

```diff
@@ -1903,3 +1903,106 @@ func (s *testPessimisticSuite) TestResolveStalePessimisticPrimaryLock(c *C) {
 
 	c.Assert(tk2.ExecToErr("admin check table t1"), IsNil)
 }
+
+func (s *testPessimisticSuite) TestIssue21498(c *C) {
```

Comment:
Need to add some `unique key uk(varchar_column)` case using the `IndexLookUp` executor fetching data from unique index and look up data in main table.

Reply:
Added snapshot read and for update read cases for `IndexLookUp`.

Reply:
How about the `insert into table select from anothertable`, we have another issue https://github.com/pingcap/tidb/issues/21506, what is the expecting behaviour for this case?